scheduler-pool/benches/scheduler.rs

```diff
@@ -0,0 +1,894 @@
+                                    pending_transactions[idx].clone(),
+                                    idx,
+                                )));
+                                // this idx can be scheduled and moved to processing
+                                processing_states[idx] = State::Processing;
+                            }
+                        }
+                        State::Processing => {
+                            is_done = false;
+                        }
+                        State::Done => {}
+                    }
+                }
+
+                if is_done {
+                    break;
+                }
+
+                let mut executor_responses: Vec<_> = vec![receiver.recv().unwrap()];
+                executor_responses.extend(receiver.try_iter());
+                for r in &executor_responses {
+                    processing_states[*signature_indices.get(r).unwrap()] = State::Done;
+                }
+            }
+            Ok(())
+        }
+    }
+
+    // frequent synchronizations hampers efficient (= parallelizable) scheduling of several chunks
+    // of txes which are tied together for each common account locks. Ideally those independent chunks can be
+    // executed in parallel, which each is consuming one worker thread as a form of serialized runs
+    // of processing. However, should synchronizations occurs between boundaries of those chunks
+    // arrival, it cannot schedule the later-coming one because it firstly flush out the the first
+    // one
+    // in other words, this is just a re-manifestation of perf. issue coming from write barriers in
+    // general.
+    fn bench_long_serialized_runs(bencher: &mut Bencher, really_synchronize: bool) {
+        let GenesisConfigInfo { genesis_config, .. } = create_genesis_config(1_000_000_000);
+        let bank = &Arc::new(Bank::new_for_tests(&genesis_config));
+        let (kp1, kp2) = (Keypair::new(), Keypair::new());
+
+        let create_tx_of_serialized_run1 = || {
+            let tx0 =
+                SanitizedTransaction::from_transaction_for_tests(system_transaction::transfer(
+                    &kp1,
+                    &solana_sdk::pubkey::new_rand(),
+                    10,
+                    genesis_config.hash(),
+                ));
+            TransactionWithIndexForBench::new((tx0, 0))
+        };
+        let create_tx_of_serialized_run2 = || {
+            let tx0 =
+                SanitizedTransaction::from_transaction_for_tests(system_transaction::transfer(
+                    &kp2,
+                    &solana_sdk::pubkey::new_rand(),
+                    10,
+                    genesis_config.hash(),
+                ));
+            TransactionWithIndexForBench::new((tx0, 0))
+        };
+
+        let _ignored_prioritization_fee_cache = Arc::new(PrioritizationFeeCache::new(0u64));
+        let pool = SchedulerPool::new(None, None, None, _ignored_prioritization_fee_cache);
```

Comment:
Leading underscores in variable names that are actually used:

```suggestion
        let prioritization_fee_cache = Arc::new(PrioritizationFeeCache::new(0u64));
        let pool = SchedulerPool::new(None, None, None, prioritization_fee_cache);
```

Unless it is important for the reader of the test to understand that the `PrioritizationFeeCache` is ignored, there seems to be no reason to add the `ignored_` prefix.

Reply:
this pr isn't still rebased onto #31319... I'll fix it when rebasing...